### ISSUE TITLE
Sample Exam Questions and Answers generation

### ISSUE THREAD

**Issue by Witiko** [danopolan/istqb_latex#85]
Using the LaTeX template, we would like to generate sample exam questions and answers from a YAML file, as discussed in https://github.com/danopolan/istqb_latex/issues/68.

Acceptance criteria:

- Create a YAML file `sample-exam-md/questions.yml` with three example questions, as discussed in https://github.com/danopolan/istqb_latex/issues/68#issuecomment-1678115505. The text of questions and the explanations will contain markdown text.
- Create a LaTeX file `sample-exam-questions.tex` that will produce a Sample Questions document from the file `sample-exam.yml`. The produced PDF document will follow the style of https://istqb-main-web-prod.s3.amazonaws.com/media/documents/ISTQB_CTFL_Sample-Exam-Questions_v4.0.pdf.
- Create a LaTeX file `sample-exam-answers.tex` that will produce a Sample Answers document from the file `sample-exam.yml`. The produced PDF document will follow the style of https://istqb-main-web-prod.s3.amazonaws.com/media/documents/ISTQB_CTFL_Sample-Exam-Answers_v4.0.pdf.

**Comment by danopolan** [comment]
Just a few updates on info from https://github.com/danopolan/istqb_latex/issues/68.

1. There is another type of question called Matching Type. Example can be seen as Question #6 in https://istqb-main-web-prod.s3.amazonaws.com/media/documents/ISTQB-CTFL-2018v3.1_Sample-Exam-B-Questions_v1.4.pdf
This question type should not change the coding but should be part of the example.
2. Reuse layout of Syllabus and Body of Knowledge documents as much as possible. Do not follow Sample Exam document linked above to the very detail. So e.g. margins, headers, footers, landing page could stay the same if content (actual text) allows it.
In general, the stress is on getting the content of questions and answers into the PDF file by refusing style we made for Syllabus and Body of Knowledge.
3. Table of content doesn't have to refer to every single question or answer. It should refer to sections only.

**Comment by Witiko** [comment]
@danopolan Thank you for the clarification. Can you also talk more about the landing page of the Sample Exam Documents? Which parts should be configurable from the `metadata.yml` file and which parts should be fixed? What is the overlap with the metadata for Syllabus and Body of Knowledge documents?

**Comment by danopolan** [comment]
Hi @Witiko,
we need to do following changes in the landing page and `metadata.yml`
1. Update `type: Syllabus` to `type: Sample Exam A – Questions` or `type: Sample Exam A – Answers`
2. Update `version: v2.0` to `version: v1.0`
3. Add text which will clarify the binding between the Questions and the Syllabus version like "Compatible with Syllabus version 4.0". This could be added as another parameter in `metadata.yml` like `compatible_syllabus: 2.0`

The landing page text for ATLaS Questions/Answers documents could be like following:
```
Certified Tester
Agile Test Leadership at Scale
(CTAL-ATLaS)
Sample Exam A – Questions
v1.0

Compatible with Syllabus version 2.0
```

**Comment by Witiko** [comment]
> 2. Reuse the layout of the Syllabus and Body of Knowledge documents as much as possible. Do not follow the Sample Exam document linked above to the very detail. So, e.g. margins, headers, footers, and landing page could stay the same if the content (actual text) allows it.
>         In general, the stress is on getting the content of questions and answers into the PDF file by refusing the style we made for the Syllabus and Body of Knowledge.

> The landing page text for ATLaS Questions/Answers documents could be like following: [...]

After comparing your description of the landing page of the sample exam documents with the current landing page of `atlas.tex` and after taking into account that headers and footers should be the same as in the syllabus and body of knowledge documents, I concluded that we want to support all metadata of the syllabus and body of knowledge documents in the sample exam documents and also support a new field `compatible_syllabus`.

I will set up an inheritance structure, so that we can have a separate theme file with definitions for sample exam documents, but most functionality will be imported from a common file with parts of the implementation that are shared between syllabus, body of knowledge, and sample exam documents.

**Comment by Witiko** [comment]
> Update `type: Syllabus` to `type: Sample Exam A – Questions` or `type: Sample Exam A – Answers`

I would omit the ` – Questions` and ` – Answers` parts, so that the metadata file can be shared between both sample exam questions and answers documents.

> The landing page text for ATLaS Questions/Answers documents could be like following:
> 
> ```
> Certified Tester
> Agile Test Leadership at Scale
> (CTAL-ATLaS)
> [...]
> ```

From the code "CT**AL**-ATLaS", I am assume that the metadata would also contain the `level: Advanced Level` field and the text "Advanced Level" should appear both below the text "Agile Test Leadership at Scale" on the landing page and in the document header?

**Comment by Witiko** [comment]
@danopolan I drafted the code for producing the landing page for Sample Exam documents (right) from `metadata.yml` (middle):

![image](https://github.com/danopolan/istqb_latex/assets/603082/e1a0a900-9190-4fd9-8fdf-d0b0afef736d)

Is this what you had in mind, @danopolan?

**Comment by Witiko** [comment]
@danopolan As discussed in https://github.com/danopolan/istqb_latex/issues/65, we cannot easily validate markdown documents. However, we can validate YAML documents against a schema. Therefore, it seems useful to add schemas for files `metadata.yml` and `questions.yml`, so that we can validate the files in the CI before typesetting the sample exam (and other) ISTQB documents.

**Comment by danopolan** [comment]
I am all for that. Will you add it as part of this issue or create a separate one?

**Comment by Witiko** [comment]
I will add it at the end of this ticket when we are content with the structure of the YAML document and the typeset output.

**Comment by Witiko** [comment]
Closed via #87.
